sklearn/preprocessing/tests/test_data.py

```diff
@@ -191,19 +210,269 @@ def test_scaler_2d_arrays():
+            else:
+                assert_equal(i * chunk_size + (batch.stop - batch.start),
+                             scaler_incr.n_samples_seen_)
+
+        # Test with 1D list
+        X = [[0., -1., 2, -0.4, 1.]]
+        scaler_incr = MinMaxScaler().partial_fit(X)
+        assert_array_almost_equal(np.array([0., -1., 2, -0.4, 1.]),
+                                  scaler_incr.data_min_)
+        assert_array_almost_equal(np.array([0., -1., 2, -0.4, 1.]),
+                                  scaler_incr.data_max_)
+        assert_array_almost_equal(np.array([1, 1, 1, 1, 1]),
+                                  scaler_incr.scale_)
+        assert_equal(1, scaler_incr.n_samples_seen_)
+
+
+def test_standard_scaler_partial_fit():
+    # Test if partial_fit run over many batches of size 1 and 50
+    # gives the same results as fit
+    rng = np.random.RandomState(0)
+    n_features = 50
+    n_samples = 1000
+
+    for chunk_size in [1, 2, 50, n_samples, n_samples + 42]:
+        offsets = rng.uniform(-1, 1, size=n_features)
+        scales = rng.uniform(1, 10, size=n_features)
+        X = rng.randn(n_samples, n_features) * scales + offsets
+
+        # Test mean at the end of the process
+        scaler_batch = StandardScaler(with_std=False).fit(X)
+
+        scaler_incr = StandardScaler(with_std=False)
+        for batch in gen_batches(n_samples, chunk_size):
+            scaler_incr = scaler_incr.partial_fit(X[batch])
+
+        assert_array_almost_equal(scaler_batch.mean_, scaler_incr.mean_)
+        assert_equal(scaler_batch.var_, scaler_incr.var_)  # Nones
+        assert_equal(scaler_batch.n_samples_seen_, scaler_incr.n_samples_seen_)
+
+        # Test std after 1 step
+        batch0 = slice(0, chunk_size)
+        scaler_incr = StandardScaler().partial_fit(X[batch0])
+        if chunk_size == 1:
+            assert_array_almost_equal(np.zeros(n_features, dtype=np.float64),
+                                      scaler_incr.var_)
+            assert_array_almost_equal(np.ones(n_features, dtype=np.float64),
+                                      scaler_incr.scale_)
+        else:
+            assert_array_almost_equal(np.var(X[batch0], axis=0),
+                                      scaler_incr.var_)
+            assert_array_almost_equal(np.std(X[batch0], axis=0),
+                                      scaler_incr.scale_)  # no constants
+
+        # Test std until the end of partial fits, and
+        scaler_batch = StandardScaler().fit(X)
+        scaler_incr = StandardScaler()  # Clean estimator
+        for i, batch in enumerate(gen_batches(n_samples, chunk_size)):
+            scaler_incr = scaler_incr.partial_fit(X[batch])
+            if batch.stop != n_samples:
+                assert_equal((i + 1) * chunk_size, scaler_incr.n_samples_seen_)
+            else:
+                assert_equal(i * chunk_size + (batch.stop - batch.start),
+                             scaler_incr.n_samples_seen_)
```

Comment:
It would be nice to make this above block as a common function, since it basically tests the same thing for all three, i.e `n_samples_seen_` is incremented in the correct way.